src/operations/query/filters/id.js

```diff
@@ -66,6 +66,84 @@ class FilterById extends BaseFilter {
         return query;
 
     }
+
+    /**
+     * Generates filter for parsedArgItem
+     * @return {import('mongodb').Filter<import('mongodb').DefaultSchema>[]}
+     */
+    filter() {
+        /**
+         * @type {import('mongodb').Filter<import('mongodb').DefaultSchema>[]}
+         */
+        const and_segments = [];
+
+        if (this.parsedArg.queryParameterValue.values) {
+            and_segments.push({
+                    $or: this.propertyObj.fields.flatMap((field) => {
+                            return {
+                                [this.parsedArg.queryParameterValue.operator]:
+                                this.filterByItems(field, this.parsedArg.queryParameterValue.values)
+                            };
+                        }
+                    ),
+                },
+            );
+        }
+
+        return and_segments;
+    }
+
+    /**
+     * Generate filter based of field and values
+     * @param {string} field
+     * @param {string[]} values
+     * @returns {Array<Object>}
+     */
+    filterByItems(field, values) {
+        if (this.enableGlobalIdSupport) {
+            const filters = [];
+            /**
+             * 3 types of values are possible
+             * 1. uuid, 2. sourceId, 3. sourceId and sourceAssigningAuthority
+             */
+            let /**@type {string[]}*/uuids = [], /**@type {string[]}*/sourceIds = [], /**@type {string[]}*/sourceIdsWithSourceAssigningAuthority = [];
+
+            values.forEach((value) => {
+                const {id, sourceAssigningAuthority} = IdParser.parse(value);
+                if (isUuid(id)) {
+                    uuids.push(id);
+                } else if (sourceAssigningAuthority) {
+                    sourceIdsWithSourceAssigningAuthority.push(value);
```

Comment:
We should convert these to UUIDs and filter on _uuid field in the query. Otherwise, _sourceAssigningAuthority column will also be used now which does not have any index.